### PR TITLE
docs(set): add documentation to the set group

### PR DESCRIPTION
**Pull request by pevisscher**

### CODE REVIEW

**Comment by pevisscher** [comment]
Current dependencies on/for this PR:
* main
  * **PR #30** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/30" target="_blank">Graphite</a> 
    * **PR #31** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/31" target="_blank">Graphite</a> 
      * **PR #32** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/32" target="_blank">Graphite</a> 
        * **PR #33** <a href="https://app.graphite.dev/github/pr/skyleague/axioms/33" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/axioms/33?utm_source=stack-comment).

File: src/set/is-disjoint/is-disjoint.ts [hunk 1]
```diff
@@ -1,6 +1,34 @@
 import { all } from '../../iterator/all'
 import type { Traversable } from '../../type'
 
+/**
+ * A disjoint set is a set of elements that have no members in common.
```
Comment (MickVanDuijn) [inline src/set/is-disjoint/is-disjoint.ts:5]:
```suggestion
 * Two sets of elements are disjoint when they have no members in common.
```

File: src/set/is-disjoint/is-disjoint.ts [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+ *
+ * The `isDisjoint` determines whether a given collection of sets are disjoint or not.
+ *
+ * There are two approaches to the problem:
+ * 1) Checking if each member of the first set is a member of the second set
+ * 2) Checking if there exists any members that are not in both sets
+ *
+ * If either approach returns true, then the sets are not disjoint and hence they are not mutually exclusive.
```
Comment (MickVanDuijn) [inline src/set/is-disjoint/is-disjoint.ts:13]:
```suggestion
 * This problem can be solved by checking if there is any member of the first set, that is also a member of the second set.
 * If there exists such a member, then the sets are not mutually exclusive.
```

File: src/set/is-subset/is-subset.ts [hunk 3]
```diff
@@ -1,8 +1,30 @@
 import { all } from '../../iterator/all'
 import type { Traversable } from '../../type'
 
-export function isSubset<T>(superset: Traversable<T>, sub: Traversable<T>) {
+/**
+ * A set is a subset of the superset if it contains all the elements in the superset.
```
Comment (MickVanDuijn) [inline src/set/is-subset/is-subset.ts:5]:
```suggestion
 * A set is a subset of the superset if all elements are contained in the superset.
```

File: src/set/is-superset/is-super.ts [hunk 4]
```diff
@@ -1,6 +1,31 @@
 import { all } from '../../iterator/all'
 import type { Traversable } from '../../type'
 
+/**
+ * A set is a superset of the subset if every element in the subset is also in the superset.
+ *
+ * We check whether each element in the superset belong to the subset.
```
Comment (MickVanDuijn) [inline src/set/is-superset/is-super.ts:7]:
```suggestion
 * We check whether each element in the subset belong to the superset.
```

**Review by MickVanDuijn — APPROVED** [review]

**Review by MickVanDuijn — CHANGES_REQUESTED** [review]
See inline comments

**Review by MickVanDuijn — APPROVED** [review]

**Comment by pevisscher** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/skyleague/axioms/33?utm_source=stack-comment).


Job Reference: K5yEkBD2dLgv1WFxOfGc

**Pull request merged**